### PR TITLE
[PW-1893] Implement Multibanco

### PR DESCRIPTION
**Pull request by cyattilakiss**

### CODE REVIEW

File: controllers/front/Payment.php [hunk 1]
```diff
@@ -514,6 +509,26 @@ private function processLocalPaymentMethod(\Cart $cart, $paymentType, $paymentIs
                     // invalid parameters
                     $isValid = false;
                 }
+                break;
+            case 'PresentToShopper':
+
+                $_SESSION['paymentAction'] = $response['action'];
+
+                $customer = new \Customer($cart->id_customer);
```
Comment (rikterbeek) [inline controllers/front/Payment.php]: should we not validate if $cart->id_customer has a value before loading it into the customer object ?

File: views/js/payment-components/local-payment-method.js [hunk 2]
```diff
@@ -24,23 +24,53 @@ jQuery(function ($) {
...
+     */
+    function mergeObjects(objectArray)
+    {
+        return objectArray.reduce(function (r, o) {
+            Object.keys(o).forEach(function (k) {
+                r[k] = o[k];
+            });
+            return r;
+        }, {});
+    }
```
Comment (rikterbeek) [inline views/js/payment-components/local-payment-method.js]: no other way possible ?

File: views/js/payment-components/stored-payment-method.js [hunk 3]
```diff
@@ -46,10 +46,8 @@ jQuery(function ($) {
     // Iterate through the stored payment methods list we got from the adyen checkout component
     checkoutStoredPaymentMethods.forEach(function (storedPaymentMethod) {
 
-        /*
-            storedPaymentMethod.id = $storedPaymentApiId in the stored-payment-method.tpl
-            if the container doesn't exits don't try to render the component
-        */
+        //  storedPaymentMethod.id = $storedPaymentApiId in the stored-payment-method.tpl
+        //  if the container doesn't exits don't try to render the component
```
Comment (rikterbeek) [inline views/js/payment-components/stored-payment-method.js]: remove this or make it the comment clear.

File: controllers/front/Payment.php (hunk 1 above)
Comment (cyattilakiss) [reply controllers/front/Payment.php]: prestashop will add an id_customer even if it is a guest user. If by any chance there is an issue with the customer id the next line: Validate::isLoadedObject($customer) will validate it.

File: views/js/payment-components/stored-payment-method.js (hunk 3 above)
Comment (cyattilakiss) [reply views/js/payment-components/stored-payment-method.js]: removed, not necessary

File: views/js/payment-components/local-payment-method.js (hunk 2 above)
Comment (cyattilakiss) [reply views/js/payment-components/local-payment-method.js]: because of IE compatibility unfortunately not.

File: adyen.php [hunk 4]
```diff
@@ -781,6 +776,20 @@ public function hookDisplayPaymentEU()
         return $payment_options;
     }
 
+    /**
+     * Retrieve the necessary default configuration parameters for a checkout component
+     *
+     * @return array
+     */
+    public function getCheckoutComponentInitData()
```
Comment (msilvagarcia) [inline adyen.php]: Can this method be private instead?

**Pull request merged**